GVRf/Framework/jni/engine/importer/assimp_importer_jni.cpp

```diff
@@ -47,4 +86,180 @@ Java_org_gearvrf_NativeAssimpImporter_getMesh(JNIEnv * env,
             reinterpret_cast<AssimpImporter*>(jassimp_importer);
     return reinterpret_cast<jlong>(assimp_importer->getMesh(index));
 }
+
+JNIEXPORT jobject JNICALL
+Java_org_gearvrf_NativeAssimpImporter_getAssimpScene(
+        JNIEnv * env, jobject obj, jlong jassimp_importer) {
+    AssimpImporter* assimp_importer =
+            reinterpret_cast<AssimpImporter*>(jassimp_importer);
+
+    jobject jassimp_scene = NULL;
+
+    const aiScene *assimp_scene = assimp_importer->getAssimpScene();
+
+    if (!assimp_scene) {
+        LOGE("import file returned null\n");
+        goto error;
+    }
+
+    if (!AssimpImporter::create_instance(env, "org/util/jassimp/AiScene", jassimp_scene)) {
+        goto error;
+    }
+
+    if (!AssimpImporter::load_scene_graph(env, assimp_scene, jassimp_scene)) {
+        goto error;
+    }
+
+    /* jump over error handling section */
+    goto end;
+
+error:
+    {
+        jclass exception = env->FindClass("java/io/IOException");
+        if (NULL == exception) {
+            /* thats really a problem because we cannot throw in this case */
+            env->FatalError("could not throw java.io.IOException");
+        }
+        env->ThrowNew(exception, aiGetErrorString());
+    }
+
+end:
+    return reinterpret_cast<jobject>(jassimp_scene);
+}
+
+JNIEXPORT jlong JNICALL
+Java_org_gearvrf_NativeAssimpImporter_getNodeMesh(JNIEnv * env,
+        jobject obj, jlong jassimp_importer, jstring jnode_name, jint index) {
+    AssimpImporter* assimp_importer =
+            reinterpret_cast<AssimpImporter*>(jassimp_importer);
+    const char *node_name = env->GetStringUTFChars(jnode_name, 0);
+    aiNode* root_node = assimp_importer->getAssimpScene()->mRootNode;
+    aiNode* current_node = root_node->FindNode(node_name);
+    env->ReleaseStringUTFChars(jnode_name, node_name);
+    return reinterpret_cast<jlong>(assimp_importer->getMesh(current_node->mMeshes[index]));
+
+}
+
+JNIEXPORT jobject JNICALL
+Java_org_gearvrf_NativeAssimpImporter_getMeshMaterial(JNIEnv * env,
+        jobject obj, jlong jassimp_importer, jstring jnode_name, jint index) {
+    AssimpImporter* assimp_importer =
+            reinterpret_cast<AssimpImporter*>(jassimp_importer);
+    const char *node_name = env->GetStringUTFChars(jnode_name, 0);
+    aiNode* root_node = assimp_importer->getAssimpScene()->mRootNode;
+
+    aiNode* current_node = root_node->FindNode(node_name);
+    env->ReleaseStringUTFChars(jnode_name, node_name);
+    return reinterpret_cast<jobject>(assimp_importer->mesh_material(env, current_node->mMeshes[index]));
+}
+
+JNIEXPORT jfloatArray JNICALL
+Java_org_gearvrf_NativeAssimpImporter_decomposeTransformationMatrix(JNIEnv * env,
+        jobject obj, jfloatArray jtransformation_matrix) {
+
+    jfloat* tm = env->GetFloatArrayElements(jtransformation_matrix,0); // Transformation Matrix
+
```

Comment:
I think you need to call ReleaseFloatArrayELments for tm before you jump out this function. Otherwise it seems to be a memory leak.